src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +65,61 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IVoidArgumentValidationConfiguration CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var fake = this.expressionParser.GetFakeManagerCallIsMadeOn(setterExpression);
+            var rule = this.ruleFactory.Invoke(setterExpression);
+
+            fake.AddRuleFirst(rule);
+
+            return this.configurationFactory.CreateConfiguration(fake, rule);
+        }
+
+        private string GetExpressionDescription<TValue>(Expression<Func<TValue>> expression)
+        {
+            var matcher = new ExpressionCallMatcher(
+                expression,
+                ServiceLocator.Current.Resolve<ExpressionArgumentConstraintFactory>(),
+                ServiceLocator.Current.Resolve<MethodInfoManager>(),
+                this.callExpressionParser);
+
+            var expressionDescription = matcher.DescriptionOfMatchingCall;
+            return expressionDescription;
+        }
+
+        private LambdaExpression BuildSetterFromGetter<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            MemberExpression memberExpression = propertySpecification.Body as MemberExpression;
+            if (memberExpression == null)

```

Comment:
> And I guess a VB test project. How terrible would it be to use the VB Integration test project for this kind of thing?

Not terrible I think, if it's an integration test ;)

Actually, the test itself could be in C#; only the class that declares an indexed property would have to be in VB (IIRC C# 4+ can consume indexed properties, even if it can't declare them)